site/plugins.md

```diff
@@ -390,8 +391,257 @@ func (p *Plugin) Report(w http.ResponseWriter, r *http.Request) {
+
+A node contains all the information about the represented object (e.g. host, container, pod, etc.).
+In particular, a node may contain:
+
+- `latest` - an id-value map containing the latest values. Each id has only one value.
+- `latestControls` - the latest available controls.
+- `metrics` - the collection of metrics to display in the UI. Each metric has multiple timestamped values.
+- `sets` - a string->set-of-strings map, for example a list of local networks.
+- `counters` - a string->int map.
+
+### Controls
+Controls describe interfaces that expose actions that the user can perform on different objects (e.g. host, container, etc.).
+Controls are an element of nodes. In this way, each control in a node is attached to it and performs an action on the object described by the node itself. Below is an example of how controls are represented in the JSON report.
+In the report, the attribute `latest_controls` contains all the controls exposed by scope and/or plugins, but only those alive will be listed in the attribute `controls`.
+
+```json
+"controls": {
+	"timestamp": "2016-11-17T08:53:04.567890059Z",
+	"controls": [
+		"switchToIOWait"
+	]
+},
+"latestControls": {
+	"switchToIOWait": {
+		"timestamp": "2016-11-17T08:53:03.171438309Z",
+		"value": {
+			"dead": false
+		}
+	},
+	"switchToIdle": {
+		"timestamp": "2016-11-17T08:53:03.171438309Z",
+		"value": {
+			"dead": true
+		}
+	}
+}
+```
+
+- `timestamp` specifies when the control was exposed.
+- `value` is an object containing the control value. At the moment, only the state is available.
+ - `dead` is a boolean to know the state (active, dead) of a control. It is useful to show controls only when they are in a usable state.
+
+### Metadata
+All metadata entries are placed within nodes in the section named `latest`.
+This section contains the latest values to display and consists of `timestamp` and `value`.
+Both should be written as json strings (with double quotes).
+Scope uses `metadata_templates` to know how to display such data.
+To pair metadata with its template, it is necessary to use the `metadata-template-id` as a key to identify that particular piece of data. Example:
+
+```json
+"metadata_templates": {
+      "metadata-templates-id": {
+        "id": "metadata-templates-id",
+        "label": "Human-readable description",
+        "priority": 1.6,
+        "from": "latest"
+      }
+}
+
+"latest": {
+	"metadata-templates-id": {
+		"timestamp": "2016-11-17T08:53:02.189193735Z",
+		"value": "42"
+	}
+}
+```
+
+### Metadata Templates
+Metadata Templates describe a particular metadata item.
+This description is used to extract metadata from a node and display it on Scope UI.
 
```

Comment:
A particular metadata template (such as "traffic-control-pktloss" in the example below) explains how to interpret the metadata with that id in all nodes.

For example, in the "host" topology, there is a metadata template "kernel_version" that teaches Scope that the kernel version should be displayed in Scope in the first position with the label "Kernel Version".

This is specified only one time in the topology. Then, all nodes can have a metadata "kernel_version" and the details of how to display it do not need to be specified again.

Metadata Templates exist so that the same information about a metadata kind do not need to be repeated in all nodes.

The word "extract" is because a metadata template does not belong to a node.

What about this:

> A metadata template describes a kind of metadata that could be present in zero, one or several nodes and it specifies how to display such metadata. Metadata templates are not placed within nodes but in the "metadata_templates" section so that the same information do not need to be repeated in all nodes.